subsys/net/openthread/Kconfig

```diff
@@ -19,12 +19,47 @@ config NORDIC_SECURITY_PROMPTLESS
 choice OPENTHREAD_SECURITY
 	depends on !OPENTHREAD_SECURITY_INTERNAL
 
+config OPENTHREAD_NRF_SECURITY_PSA_CHOICE
+	bool "nRF Security with PSA crypto enabled"
+	select OPENTHREAD_NRF_SECURITY_PSA
+
 config OPENTHREAD_NRF_SECURITY_CHOICE
 	bool "nRF Security"
 	select OPENTHREAD_NRF_SECURITY
 
 endchoice
 
+config OPENTHREAD_NRF_SECURITY_PSA
+	bool
+	select EXPERIMENTAL
+	select NRF_SECURITY
+	select MBEDTLS_ENABLE_HEAP
+	select MBEDTLS_PSA_CRYPTO_STORAGE_C if !BUILD_WITH_TFM
+	select PSA_NATIVE_ITS if !BUILD_WITH_TFM
+	select PSA_WANT_ALG_RSA_OAEP
+	select PSA_WANT_ALG_HKDF
+	select PSA_WANT_ALG_CCM
```

Comment:
Is CCM needed? Note that OpenThread implements CCM itself and only AES_ECB_NO_PADDING is required from the crypto layer. HKDF is also implemented in OpenThread with HMAC.

Reply:
These configurations are mainly based on sample provided by crypto team:
https://github.com/nrfconnect/sdk-nrf/blob/main/samples/crypto/psa_tls/prj.conf
Also from config in openthread:
https://github.com/openthread/openthread/blob/main/third_party/mbedtls/mbedtls-config.h

In both files CCM support is defined. Also PSA TLS sample enables HKDF. 
@frkv could you confirm if configurations here are correct and needed?

Reply:
Sample from the crypto team doesn't really tell us anything about OT requirements, but the reference mbedTLS config indeed defines MBEDTLS_TLS_ECJPAKE_WITH_AES_128_CCM_8, so looks like AES CCM may be required for TLS, so let's keep it :).